### PR TITLE
background support for save issue filing settings 

### PR DESCRIPTION
**Pull request by haonliu**
#### Description of changes

Background support for save issue filing settings. This will be used in Wahaj's change to fix the cancel button's UX.
No UX changes.

#### Pull request checklist

- [x] WI - 1502923
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [x] Added screenshots/GIFs for UI changes.

### CODE REVIEW

File: src/background/stores/global/user-configuration-store.ts [hunk 1]
```diff
@@ -86,6 +88,14 @@ export class UserConfigurationStore extends BaseStoreImpl<UserConfigurationStore
         this.saveAndEmitChanged();
     }
 
+    @autobind
+    private onSaveIssueSettings(payload: SaveBugFilingSettingsPayload): void {
+        const bugService = payload.bugServiceName;
+        this.state.bugService = payload.bugServiceName;
```
Comment (waabid) [inline src/background/stores/global/user-configuration-store.ts]: You can reuse your variable 'bugService' here.

File: src/background/actions/action-payloads.ts [hunk 2]
```diff
@@ -128,6 +129,10 @@ export interface SetBugServicePayload extends BaseActionPayload {
     bugServiceName: string;
 }
 
+export interface SaveBugFilingSettingsPayload extends SetBugServicePayload {
```
Comment (peterdur) [inline src/background/actions/action-payloads.ts]: Should this be Save_Issue_FilingSettingsPayload given that the message is `SaveIssueFilingSettings`

Comment (haonliu) [reply src/background/actions/action-payloads.ts]: We have a work item (1506355) to rename all at once. In this PR there are several places that I don't really know which one should I use at present.

File: src/background/actions/action-payloads.ts [hunk 3]
```diff
@@ -128,6 +129,10 @@ export interface SetBugServicePayload extends BaseActionPayload {
     bugServiceName: string;
 }
 
+export interface SaveIssueFilingSettingsPayload extends SetBugServicePayload {
```
Comment (peterdur) [inline src/background/actions/action-payloads.ts]: Having read more of the code now, and particularly thinking about the `extends SetBugServicePayload` here, I'm thinking that my earlier suggestion about renaming went the wrong way. It looks like this message should have the same semantics as `SetBugService` but also set the bundle of properties. If that's correct, I would suggest the following:

- Message to be named `SetBugServiceAndProperties`
- Payload to be named `SetBugServiceAndPropertiesPayload`
- Property within payload to be named `bugServiceProperties` to match type `BugServiceProperties`
- Methods related to `SetBugServiceAndProperties` should immediately follow their `SetBugService` analogs.

Thoughts?

Comment (haonliu) [reply src/background/actions/action-payloads.ts]: Make sense!

**Pull request merged**